docs: modelix release 24.1 (#12)
modelix/website: content/en/blog/2024-05-16-modelix-platform-release-24.1/index.md

---
date: 2024-05-16
title: "Modelix Platform Release 24.1"
---

Version **`24.1`** of the modelix platform is out now!

The versions bundled in this release are:

```
modelixCoreVersion = 7.4.0
incrementalVersion = 0.2.0
buildToolsVersion = 1.5.0

> mpsPluginsVersion = 0.7.0

Reviewer: Wouldn't it be easier for the users to have actual artifact names (and links to some repo) here instead of these abstract versions?

Reviewer 2: `mpsPluginsVersion` shall be updated if https://github.com/modelix/modelix.mps-plugins/pull/100 is merged before release. (Which I suggest to happen.)

Author: @languitar it's the way we've always done it, we can discuss this for future releases.

Reviewer: Why not improving this right now?

Author: I added a link that will point to the versions toml, once it is released. 
If someone really needs the information that detailed then that should suffice.

```

> If you want to know how to use this platform in your Gradle build, check out the corresponding [documentation](https://docs.modelix.org/modelix/23.2/platform/howto/usage-platform.html).

Reviewer: ```suggestion
If you want to know how to use this platform in your Gradle build, check out the [corresponding usage documentation](https://docs.modelix.org/modelix/23.2/platform/howto/usage-platform.html).
```

- [Feature Highlights](#feature-highlights)
  - [Support for newer MPS Versions](#support-for-newer-mps-versions)

>   - [Stabilization of Bulk Model Synchronization Plugin](#stabilization-of-bulk-model-synchronization-plugin)

Reviewer: ```suggestion
  - [Stabilization of the Bulk Model Synchronization Plugin](#stabilization-of-bulk-model-synchronization-plugin)
```

- [Development Insights](#development-insights)
  - [MPS Sync Plugin (experimental)](#mps-sync-plugin-experimental)

>   - [Adapter for LionWeb Bulk API (experimental)](#adapter-for-lionweb-bulk-api-experimental)

Reviewer: ```suggestion
  - [Adapter for the LionWeb Bulk API (experimental)](#adapter-for-lionweb-bulk-api-experimental)
```

  - [Modelix LinkedIn Page](#modelix-linkedin-page)

# Feature Highlights

## Support for newer MPS Versions

> Modelix components are now compatible with MPS 2023.2 and below.

Reviewer 2: Maybe we could make this sentence mor explicit: `Modelix components are now compatible with MPS 2021.1 - 2023.2.`

> To decrease the required maintenance effort, we decided to drop support for MPS 2020.3.

Reviewer: ```suggestion
To reduce the required maintenance effort, we decided to drop support for MPS 2020.3.
```

While some use cases might still work with 2020.3, we highly recommend using a more recent MPS version.

> ## Stabilization of Bulk Model Synchronization Plugin

Reviewer: ```suggestion
## Stabilization of the Bulk Model Synchronization Plugin
```

> The [bulk-model-sync Gradle Plugin](https://docs.modelix.org/modelix/23.2/core/reference/component-bulk-model-sync-gradle.html) allows bulk synchronization of models between MPS and a [model-server](https://docs.modelix.org/modelix/23.2/core/reference/component-model-server.html).

Reviewer: These links should probably point to the new version once available.

> The [bulk-model-sync Gradle Plugin](https://docs.modelix.org/modelix/23.2/core/reference/component-bulk-model-sync-gradle.html) allows bulk synchronization of models between MPS and a [model-server](https://docs.modelix.org/modelix/23.2/core/reference/component-model-server.html).

Reviewer: ```suggestion
The [bulk-model-sync Gradle Plugin](https://docs.modelix.org/modelix/23.2/core/reference/component-bulk-model-sync-gradle.html) allows synchronizing models between MPS and a [model-server](https://docs.modelix.org/modelix/23.2/core/reference/component-model-server.html).
```

It's either "allows someone to synchronize" or allows + gerund (i.e. -ing).

> The [bulk-model-sync Gradle Plugin](https://docs.modelix.org/modelix/23.2/core/reference/component-bulk-model-sync-gradle.html) allows bulk synchronization of models between MPS and a [model-server](https://docs.modelix.org/modelix/23.2/core/reference/component-model-server.html).

Reviewer 2: ```suggestion
The [bulk-model-sync Gradle Plugin](https://docs.modelix.org/modelix/23.2/core/reference/component-bulk-model-sync-gradle.html) allows the bulk synchronization of models between MPS and a [model-server](https://docs.modelix.org/modelix/23.2/core/reference/component-model-server.html).
```

> It is now considered stable, but it is still actively being worked with regards to performance.

Reviewer: ```suggestion
The plugin is now stable.
However, we continue improving its performance in the future.
```

> # Development Insights

Reviewer: We should add a sentence what this means and what type of content we present here.

Author: I restructured it a bit. If the heading needed further explanation, it was not a good heading.

## MPS Sync Plugin (experimental)

> We are hard at work to create a new Kotlin-based MPS Sync Plugin, which will allow you to bind local MPS repositories to remote model-server repositories.

Reviewer: "hard at work" sounds a bit fishy

Reviewer 2: "A new MPS Sync Plugin is in the making, which ..." might sound a bit better.

You will be able to seamlessly exchange model data between MPS and the model-server.

> Changes performed on one side will be automatically synchronized to the other side.

Reviewer: This text could require a sentence on how this approach differs from the bulk sync plugin.

Reviewer 2: And to the legacy sync plugin. 

By the way, have we talked about the legacy sync plugin as a standalone library in any of the release notes recently?

Author: I don't think so @benedekh. Could you try and put the difference between legacy and new sync plugin into words? You're more familiar with it than I am.

Reviewer 2: I'm not that much familiar with the legacy plugin. Let's postpone it until the official release of the new sync plugin.

Reviewer: We've clarified this a bit.

> ## Adapter for LionWeb Bulk API (experimental)

Reviewer: ```suggestion
## New Adapter for the LionWeb Bulk API (experimental)
```

> ## Adapter for LionWeb Bulk API (experimental)

Reviewer: Should we add a sentence on what LionWeb is for readers not knowing it?

A first version of the [lionweb-modelix-adapter](https://github.com/LionWeb-io/lionweb-modelix-adapter) has been made available on the LionWeb GitHub.
It allows using the model-server with the LionWeb Bulk API, but comes with limitations detailed in the README.
We recommend using the modelix model-api to make use of advanced model-server features such as real-time collaboration.

## Modelix LinkedIn Page

> We created a [Modelix LinkedIn Page](https://www.linkedin.com/company/modelixweb/), where we are regularly sharing Modelix related content including additional development insights.

Reviewer: ```suggestion
We created a [Modelix LinkedIn Page](https://www.linkedin.com/company/modelixweb/) where we are regularly sharing Modelix-related content including additional development insights.
```

---
date: 2024-05-16
title: "Modelix Platform Release 24.1"
---

Version **`24.1`** of the modelix platform is out now!
The platform is now independent of the MPS version and works with all supported MPS versions.
Simply use `org.modelix:platform:24.1.0` in your build scripts.

The versions bundled in this release are:

```
modelixCoreVersion = 7.4.0
incrementalVersion = 0.2.0
buildToolsVersion = 1.5.0

> mpsPluginsVersion = 0.7.0

Reviewer 2: ```suggestion
mpsPluginsVersion = 0.7.1
```

MPS Plugins 0.7.1 were released: https://github.com/modelix/modelix.mps-plugins/releases/tag/0.7.1
